python/cudf/cudf/core/tools/numeric.py

```diff
@@ -212,13 +223,19 @@ def _convert_str_col(col, errors, _downcast=None):
     if not is_string_dtype(col):
         raise TypeError("col must be string dtype.")
 
-    is_integer = libstrings.is_integer(col)
-    if is_integer.all():
-        return col.astype(dtype=cudf.dtype("i8"))
+    if col.is_integer().all():
+        return col.astype(dtype=cudf.dtype("i8"))  # type: ignore[return-value]
 
-    col = _proc_inf_empty_strings(col)
+    converted_col = (
```

Comment:
Should we keep this TODO?
```suggestion
    # TODO: This can be handled by libcudf in
    # future see StringColumn.as_numerical_column
    converted_col = (
```

Reply:
Good point. Yes, let's keep this comment